tests/unit-tests/test_dataloading.py

```diff
@@ -453,6 +453,88 @@ def test_GSgnnData2():
+        # ('n0', 'r1', 'n1'), ('n0', 'r0', 'n1'), ("n1", "r2", "n0")
+        gdata = GSgnnData(part_config=part_config)
+        
+        # Test 0: empty edge feature fields, should return an empty list
+        efeat_fields = {}
+        target_idx = {('n0', 'r1', 'n1'): [0]}
+        dataloader = GSgnnEdgeDataLoader(gdata, target_idx, [100, 100], 10,
+                                        label_field='label',
+                                        train_task=False, remove_target_edge_type=False)
+        for _, _, blocks in dataloader:
+            edge_feat_list = gdata.get_blocks_edge_feats(blocks, efeat_fields)
+            assert len(edge_feat_list) == 0
+
+        # Test 1: all edge feature fields, should return a list with two dicts
+        efeat_fields = {
+            ('n0', 'r0', 'n1'): ['feat'],
+            ('n0', 'r1', 'n1'): ['feat']
+        }
+        for _, _, blocks in dataloader:
+            edge_feat_list = gdata.get_blocks_edge_feats(blocks, efeat_fields)
+            assert len(edge_feat_list) == 2
+            assert edge_feat_list[0][('n0', 'r0', 'n1')].shape[1] == 2
+            assert edge_feat_list[0][('n0', 'r1', 'n1')].shape[1] == 2
+            assert edge_feat_list[1][('n0', 'r1', 'n1')].shape[1] == 2
+            assert_equal(edge_feat_list[1][('n0', 'r1', 'n1')][0].numpy(),
+                         gdata.g.edges[('n0', 'r1', 'n1')].data['feat'][0][0].numpy())
+
+        # Test 2: partial edge feature fields, should return a list with two dicts
+        efeat_fields = {
+            ('n0', 'r1', 'n1'): ['feat']
+        }
+        for _, _, blocks in dataloader:
+            edge_feat_list = gdata.get_blocks_edge_feats(blocks, efeat_fields)
+            assert len(edge_feat_list) == 2
+            assert edge_feat_list[0][('n0', 'r1', 'n1')].shape[1] == 2
+            assert edge_feat_list[1][('n0', 'r1', 'n1')].shape[1] == 2
+            assert_equal(edge_feat_list[1][('n0', 'r1', 'n1')][0].numpy(),
+                         gdata.g.edges[('n0', 'r1', 'n1')].data['feat'][0][0].numpy())
+
+        # Test 3: one non-existing edge feature fields, will trigger an assertion error
+        efeat_fields = {
+            ('n0', 'r0', 'n1'): ['none'],
+            ('n0', 'r1', 'n1'): ['feat', 'none']
+        }
+        for _, _, blocks in dataloader:
+            try:
+                edge_feat_list = gdata.get_blocks_edge_feats(blocks, efeat_fields)
+            except:
+                edge_feat_list = {}
+            assert len(edge_feat_list) == 0
+
+        # Test 4: all non-existing edge feature fields, should trigger an assertion error
+        efeat_fields = {
+            ('n0', 'r0', 'n1'): ['none'],
+            ('n0', 'r1', 'n1'): ['none']
+        }
+        for _, _, blocks in dataloader:
+            try:
+                edge_feat_list = gdata.get_blocks_edge_feats(blocks, efeat_fields)
+            except:
+                edge_feat_list = {}
+            assert len(edge_feat_list) == 0
+
```

Comment:
Can we also test the case when there are two edge features and both exists in the graph? And also the case when layer 0 does not have edge feature but layer 1 does?

Reply:
added two new test. Test case 5 in `test_GSgnnData_edge_feat()`, and Test case 1 in `test_GSgnnData_edge_feat2()`